aws-sdk-go/LeoFSTest.go

```diff
@@ -0,0 +1,485 @@
+    res, err := svc.GetObject(&s3.GetObjectInput{
+        Bucket  : bucketStr,
+        Key     : keyStr,
+        Range   : rangeStr,
+    })
+    if err != nil {
+        log.Fatalln(err)
+    }
+    reader, _ := os.Open(path)
+    reader.Seek(int64(start), 0)
+    tmp := make([]byte, end - start + 1)
+    io.ReadFull(reader, tmp)
+    buf, _ := ioutil.ReadAll(res.Body)
+    if !bytes.Equal(tmp, buf) {
+        log.Fatalln("Content NOT Match!")
+    }
+
+    log.Println("===== Range Get Object End =====")
+    log.Println()
+}
+
+func copyObject(bucketName, src, dst string) {
+    log.Printf("===== Copy Object [%s/%s] -> [%s/%s] Start =====\n",
+        bucketName, src, bucketName, dst)
+    sourceStr := aws.String(bucketName + "/" + src)
+    bucketStr := aws.String(bucketName)
+    dstStr := aws.String(dst)
+
+    _, err := svc.CopyObject(&s3.CopyObjectInput{
+        CopySource  : sourceStr,
+        Bucket  : bucketStr,
+        Key     : dstStr,
+    })
+    if err != nil {
+        log.Fatalln(err)
+    }
+
+    log.Println("===== Copy Object End =====")
+    log.Println()
+}
+
+func listObject(bucketName, prefix string, expected int) {
+    log.Printf("===== List Objects [%s/%s*] Start =====\n", bucketName, prefix)
+    bucketStr := aws.String(bucketName)
+    prefixStr := aws.String(prefix)
+    res, err := svc.ListObjects(&s3.ListObjectsInput{
+        Bucket  : bucketStr,
+        Prefix  : prefixStr,
+    })
+    if err != nil {
+        log.Fatalln(err)
+    }
+    count := 0
+    for _, obj := range res.Contents {
+        if doesFileExist(bucketName, *obj.Key) {
+            log.Printf("%s \t Size: %d\n", *obj.Key, *obj.Size)
+            count ++
+        }
+    }
+    if expected >= 0 {
+        if count != expected {
+            log.Fatalln("Number of Objects NOT Match!")
+        }
+    }
+
+    log.Println("===== List Objects End =====")
+    log.Println()
+}
+
+func deleteAllObjects(bucketName string) {
+    log.Printf("===== Delete All Objects [%s] Start =====\n", bucketName)
+    bucketStr := aws.String(bucketName)
+    res, err := svc.ListObjects(&s3.ListObjectsInput{
+        Bucket  : bucketStr,
+    })
+    if err != nil {
+        log.Fatalln(err)
+    }
+    for _, obj := range res.Contents {
+        _, errd := svc.DeleteObject(&s3.DeleteObjectInput{
+            Bucket  : bucketStr,
+            Key     : obj.Key,
+        })
+        if errd != nil {
+            log.Fatalln(errd)
+        }
+    }
+
+    log.Println("===== Delete All Objects End =====")
+    log.Println()
+}
+
+func putDummyObjects(bucketName, prefix string, total int, holder string) {
+    bucketStr := aws.String(bucketName)
+    for i := 0; i < total; i++ {
+        reader, _ := os.Open(holder)
+        svc.PutObject(&s3.PutObjectInput{
+            Bucket  : bucketStr,
+            Key     : aws.String(fmt.Sprintf("%s%d", prefix, i)),
+            Body    : reader,
+        })
+        reader.Close()
```

Comment:
Change `defer reader.Close()` and place it at the next line of `reader, _ := os.Open(holder)`